dist/mapdc.js

```diff
@@ -48807,20 +48837,36 @@ function setColorScaleDomain_v1(domain) {
   };
 }
 
-function getLegendStateFromChart(chart, useMap) {
+function getLegendStateFromChart(chart, useMap, selectedLayer) {
+  var legend = document.getElementsByClassName('legend');
+  var layers = chart.getLayerNames();
+
+  if (legend.length > layers.length && selectedLayer && selectedLayer.currentLayer !== "master") {
+    for (var i = legend.length - 1; i >= 0; --i) {
+      if (i !== selectedLayer.currentLayer) {
+        legend[i].remove();
+      }
+    }
+  }
   return toLegendState(chart.getLayerNames().map(function (layerName) {
     var layer = chart.getLayer(layerName);
+    var layerState = layer.getState();
     var color = layer.getState().encoding.color;
-    if (_typeof(color.scale) === "object" && color.scale.domain === "auto") {
-      return _extends({}, color, {
-        scale: _extends({}, color.scale, {
+
+    if (layers.length > 1 || _.isEqual(selectedLayer, layerState)) {
+      if (_typeof(color.scale) === "object" && color.scale.domain === "auto") {
```

Comment:
Just a note: I know that's just how the code was before, but I'm wondering what `_typeof()` is doing that `typeof` can't do. And `typeof` is used just a few line under.